pkg/controller/resourcegraphdefinition/controller.go

```diff
@@ -37,79 +34,89 @@ import (
 
 // ResourceGraphDefinitionReconciler reconciles a ResourceGraphDefinition object
 type ResourceGraphDefinitionReconciler struct {
-	log        logr.Logger
-	rootLogger logr.Logger
-
 	allowCRDDeletion bool
 
+	// Client and instanceLogger are set with SetupWithManager
+
 	client.Client
+	instanceLogger logr.Logger
+
 	clientSet  *kroclient.Set
 	crdManager kroclient.CRDClient
 
-	metadataLabeler   metadata.Labeler
-	rgBuilder         *graph.Builder
-	dynamicController *dynamiccontroller.DynamicController
+	metadataLabeler         metadata.Labeler
+	rgBuilder               *graph.Builder
+	dynamicController       *dynamiccontroller.DynamicController
+	MaxConcurrentReconciles int
```

Comment:
shall we make this private?

Reply:
I would rather make everything public and remove the java getter/setter/constructors, but for now I can make it private.

Reply:
I very rarelly do getters/setters, but `New` constructures are quiet common in go